From ca506a2d39152738907db63303985fe049e45d2a Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Mon, 22 Nov 2021 13:46:33 +0000
Subject: [PATCH 3/3] Enable ViPER4Android FX (2/2)

Only enabled if persist.sys.ponces.enable_v4afx is set to true

---
 app/src/main/java/me/phh/treble/app/Misc.kt         | 4 ++++
 app/src/main/java/me/phh/treble/app/MiscSettings.kt | 1 +
 app/src/main/res/xml/pref_misc.xml                  | 4 ++++
 3 files changed, 9 insertions(+)

diff --git a/app/src/main/java/me/phh/treble/app/Misc.kt b/app/src/main/java/me/phh/treble/app/Misc.kt
index e2471cf..b4806d8 100644
--- a/app/src/main/java/me/phh/treble/app/Misc.kt
+++ b/app/src/main/java/me/phh/treble/app/Misc.kt
@@ -228,6 +228,10 @@ object Misc: EntryStartup {
                     asusSvc.setDclickEnable(if(value) 1 else 0)
                 }
             }
+            MiscSettings.enableV4afx -> {
+                val value = sp.getBoolean(key, false)
+                SystemProperties.set("persist.sys.ponces.enable_v4afx", if (value) "1" else "0")
+            }
         }
     }
 
diff --git a/app/src/main/java/me/phh/treble/app/MiscSettings.kt b/app/src/main/java/me/phh/treble/app/MiscSettings.kt
index c1dd369..60be639 100644
--- a/app/src/main/java/me/phh/treble/app/MiscSettings.kt
+++ b/app/src/main/java/me/phh/treble/app/MiscSettings.kt
@@ -35,6 +35,7 @@ object MiscSettings : Settings {
     val minimalBrightness = "key_misc_minimal_brightness"
     val aod = "key_misc_aod"
     val dt2w = "key_misc_dt2w"
+    val enableV4afx = "key_misc_enable_v4afx"
 
     override fun enabled() = true
 }
diff --git a/app/src/main/res/xml/pref_misc.xml b/app/src/main/res/xml/pref_misc.xml
index 08fd209..81edeae 100644
--- a/app/src/main/res/xml/pref_misc.xml
+++ b/app/src/main/res/xml/pref_misc.xml
@@ -28,6 +28,11 @@
 			android:defaultValue="false"
 			android:key="key_misc_disable_audio_effects"
 			android:title="Disable audio effects" />
+		<CheckBoxPreference
+			android:defaultValue="false"
+			android:key="key_misc_enable_v4afx"
+			android:title="Enable ViPER4Android FX"
+			android:summary="Requires a reboot" />
 	</PreferenceCategory>
 	<PreferenceCategory android:title="Display">
 		<EditTextPreference
-- 
2.34.0

